src/components/Initiatives/InitiativeDetailsPage/InitiativeDetailsPage.tsx

```diff
@@ -42,28 +50,76 @@ enum InitiativeDetailsTab {
   'Rules' = 'Rules',
 }
 
-export const InitiativeDetailsPage = () => {
-  const { id: initiativeId } = useRouteRefParams(initiativeRouteRef);
-  const cortexApi = useApi(cortexApiRef);
+export interface InitiativeDetailsPageProps {
+  initiative: InitiativeWithScores;
+  actionItems: InitiativeActionItem[];
+  entitiesByTag: Record<HomepageEntity['codeTag'], HomepageEntity>;
+}
 
+export const InitiativeDetailsPage: React.FC<InitiativeDetailsPageProps> = ({
+  actionItems,
+  entitiesByTag,
+  initiative,
+}) => {
   const [activeTab, setActiveTab] = useState(InitiativeDetailsTab.Failing);
   const [isFilterDialogOpen, setIsFilterDialogOpen] = useState(false);
 
-  // Have to store lambda of lambda for React to not eagerly invoke
-  const [filter, setFilter] = useState<() => Predicate<string>>(
-    () => () => true,
-  );
+  const { filters, oneOf } = useFiltersFromQueryParams(window.location.search);
```

Comment:
We probably shouldn't be accessing window in render, probably should be using one of the navigation/location hooks to grab search

Reply:
Ah whoops good call -- forgot to switch this out after testing to make sure the functionality worked 👍

Reply:
Updated ✔️